src/app/debug/table/table.component.ts

```diff
@@ -215,10 +219,31 @@ export class TableComponent implements OnInit, OnDestroy {
       next: (settings) => {
         this.tableSettings.reportsInProgress = settings.reportsInProgress;
         this.tableSettings.estimatedMemoryUsage = settings.estMemory;
+        this.loadReportInProgressDates();
       },
     });
   }
 
+  loadReportInProgressDates() {
+    let hasChanged: boolean = false;
+    for (let i = 1; i <= this.tableSettings.reportsInProgress; i++) {
+      this.httpService.getReportInProgress(i).subscribe((report: Report) => {
+        if (this.reportsInProgress[report.correlationId] == null) {
+          this.reportsInProgress[report.correlationId] = report.startTime;
+        } else if (
+          Date.now() - new Date(this.reportsInProgress[report.correlationId]).getTime() >
+          this.reportsInProgressThreshold
```

Comment:
It might be nice to encapsulate this `if` to make it better readable what it is.

e.g. `if(this.reportIsInTreshosld(report))`